Add post inference feature -  new chainsaw model

EDsteve opened the pull request:
Check the [README-post-inference-feature.md](https://github.com/LIFsCode/ELOC-3.0/commit/cfea73d142cd826ca2d16278c3fb619bd818f588#diff-7d13634f324ac6f89feee31c3a51cebcb1a30bbfd60bddcc042eb62c7feb5f80) for more details.

EDsteve committed:
Update README-ai.md

EDsteve committed:
chainsaw model

EDsteve committed:
Chainsaw model added

EDsteve committed:
Merge branch changes - updated README-ai.md

EDsteve committed:
Post-Inference feature and  settings added

EDsteve committed:
Added LoRa session persistence

EDsteve committed:
Added missing rtc_lorawan_session_t struct definitions

EDsteve committed:
Fixed DevNonces not loaded from NVS. Fixed GPIO errors at boot when LoRa disabled.

LIFsCode commented on eloc610LowPowerPartition/lib/edge-impulse/src/EdgeImpulse.cpp:
```diff
@@ -250,3 +251,56 @@ void EdgeImpulse::updateEventInfo(const ei_impulse_result_classification_t* resu
+    detectionWindow.writeIndex = (detectionWindow.writeIndex + 1) % 128;
+    
+    // Update count (max 128)
+    if (detectionWindow.count < 128) {
```
Die 128 würd ich als define/constant im Header efinieren, dann können wir das leichter erweitern.

LIFsCode commented:
Ich habe bei config änderungen immer noch im root die elocConfig.json angepast auf die aktuellen defaults.

Sonst sind die Änderungen gut, können so rein

EDsteve committed:
Added LoRa Join audio indicator via buzzer

EDsteve committed:
fix(lorawan): auto-rejoin on NOT_JOINED with rate-limited retry. Added log reset reason at boot

EDsteve committed:
Disable dwell time limits for AS923

EDsteve committed:
feat: Add PUI DMM-4026-B-I2S-R microphone support

- Add microphone type selection in project_config.h

- Add DMM-4026-B-I2S-R specific I2S configuration (MCLK 256x, APLL)

- Add DC offset removal filter for DMM-4026-B-I2S-R (18-bit precision)

- Add microphone identification logging at startup

- Add documentation (README-DMM-4026-B-I2S-R.md)

EDsteve committed:
Fix: I2S tTiming issues with DMM_4026_B Microphone

EDsteve committed:
Add memory-bank and .clinerules

EDsteve committed:
fix: Delay AI-startup for stable BT connections - correct NVS.bin path
